numpy/random/tests/test_random.py

```diff
@@ -209,6 +243,38 @@ def test_repeatability(self):
         res = hashlib.md5(val).hexdigest()
         assert_(tgt[np.dtype(np.bool).name] == res)
 
+    def test_int64_uint64_corner_case(self):
+        # When stored in Numpy arrays, `lbnd` is casted
+        # as np.int64, and `ubnd` is casted as np.uint64.
+        # Checking whether `lbnd` >= `ubnd` used to be
+        # done solely via direct comparison, which is incorrect
+        # because when Numpy tries to compare both numbers,
+        # it casts both to np.float64 because there is
+        # no integer superset of np.int64 and np.uint64. However,
+        # `ubnd` is too large to be represented in np.float64,
+        # causing it be round down to np.iinfo(np.int64).max,
+        # leading to a ValueError because `lbnd` now equals
+        # the new `ubnd`.
+
+        dt = np.int64
+        tgt = np.iinfo(np.int64).max
+        lbnd = np.int64(np.iinfo(np.int64).max)
+        ubnd = np.uint64(np.iinfo(np.int64).max + 1)
+
+        # None of these function calls should
+        # generate a ValueError now
+        actual = np.random.randint(lbnd, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, ubnd, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint(lbnd, [ubnd] * 3, dtype=dt)
+        assert_equal(actual, tgt)
+
+        actual = np.random.randint([lbnd] * 3, [ubnd] * 3, dtype=dt)
```

Comment:
I think that all of these should error with `lbnd and ubnd should be the same time` or `cannot coerce combined type of `int64` and `uint64` to fit into a single integral type`.

Note that these tests wouldn't have passed before either - so I think deciding how to expose asking for random numbers at the extremes of their bounds is best left for another PR. I can think of a bunch of possibly more desirable alternatives (with `i64max = np.iinfo(np.int64).max`):

* `np.random.randint( np.int64(i64max),  np.int64(i64max) + 1, wrap=True)
    Would also support things like 
* `np.random.randint(  np.int64(i64max),   np.int64(i64max)), closed=True)